### ISSUE TITLE
[web] SortOrder query is slow 

### ISSUE THREAD

**Issue by ductranit** [tekartik/sembast.dart#189]
My table has about 1000 records. When I use `Sembast.SortOrder` the query response is very slow (about 15 - 30 seconds)
My query is just simple:
```
var finder = Sembast.Finder();
finder.sortOrder = Sembast.SortOrder("updatedAt", true);
var records = await store.find(db, finder: finder);
```
When I remove the sort & query all, it's very fast (less than 1 second)
The problem is that this issue isn't always happened (mostly I see it when I refresh a page) 
Do you have any idea to fix it?
My system:
`Flutter (Channel master, 1.20.0-2.0.pre, on Microsoft Windows [Version 10.0.19041.329], locale en-US)`

**Comment by alextekartik** [comment]
Thanks for the report.
Indeed that seems excessive. Does it happen when you run normally or when you debug the application? My guess is that it happens during debugging only but I would be wrong. One thing you can try:
call `disableSembastCooperator()` in your main in debug mode before any database operation to see if it solves your issue.

Since you're not the first one reporting this issue (I guess my work linux pc is fast), I might change its behavior in web mode.

**Comment by ductranit** [comment]
hi @alextekartik, this issue happens on my production site as well (I built with `flutter build web`), especially when I press refresh the site, it loads very slow, then it works normally until I press refresh again.
And weird that only the sort query has this issue. Other queries (equal, matches...) work normally.
I tried `disableSembastCooperator()` in debug and it works. But the question is this method safe for production? I see this warning:
```
The member 'disableSembastCooperator' can only be used within 'package:sembast/src/cooperator.dart' or a test.dart
```

**Comment by alextekartik** [comment]
Basically this weird [cooperator](https://github.com/tekartik/sembast.dart/blob/master/sembast/doc/implementation_details.md#cooperator) was added after some device testing in native mode. Since everything runs in on thread, sorting 10K+ records using regular sort method and a complex compare function, jank could happen. What it adds is a pause to let other async method (and drawing) work. The timing (pause 100us every 4ms) was optimized for native code with and it seems web is a lot slower causing too many pauses and exponential delay. I'm pretty sure I can optimize the delay slightly differently on the web to avoid this huge delay.

In the meantime, disableSembastCooperator() is safe (the warning is just a meta information since I only use it for performance testing in test mode). And if you don't see any jank you can keep it in production. Unfortunately I don't get this issue on my linux machine (i7 8700K, 32Gb) on the web (I have a test app with 2K+ records that sorts ok) and I don't have a good reproduction case at this point.

**Comment by ductranit** [comment]
Thank for your help.
It looks like this issues only happens on windows machine. My client says that it runs normally on chrome mac os.
It's good to hear that I can use `disableSembastCooperator()` in production.
I'm happy to help when you release a dev version to optimize the delay for web. Just ping me on this thread when the new build is available.
P/s: I tested on the release build and it seems good to go. The query speed is fast. I'll let you know if I find other issue relates to this one.

**Comment by alextekartik** [comment]
I published a new sembast version (2.4.6) that specifies different delay for the cooperator. It still impact performance (about 20% in my testing but not 1500% like you noticed before) to the benefit of no jank (that could be seen while scrolling a huge list).

I would be interested in whether that works for you (without calling disableSembastCooperator). As note, you can now specify custom timing in enableSembastCooperator, although my wish is that nobody needs that.

**Comment by ductranit** [comment]
I tested version 2.4.6 without calling `disableSembastCooperator()` and saw that the result is looking good on chome windows.
The query runs really fast on both debug & release so I don't need to use `enableSembastCooperator` to custom timing.
Nice job!
